### PR TITLE
Fix test_neighbor_mac on t1-lag and t1-64-lag

### PR DESCRIPTION
**Pull request by byu343**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [x] 202311
- [x] 202405

#### What is the motivation for this PR?
A recent change applies this test to the topo of t1 and t1-variants:
https://github.com/sonic-net/sonic-mgmt/pull/12952

The test can pass on topo t1 without any change.

For t1-lag or t1-64-lag, the Ethernet0 to test is in portchannel, so IP address cannot be set to it. 

#### How did you do it?
The fix here is to remove the intf from the portchanenl before testing it.
Skip the test if Ethernet0 is not available or down at the beginning of test
Assert Ethernet0 is in routed state after removing it from portchannel 

#### How did you verify/test it?
Run the test on t1-lag and t1-64-lag topo.
 - The test can pass
 - Verified in the log that removing and adding the member to PortChannel succeeded

### CODE REVIEW

**Review by wsycqyz — APPROVED** [review]

File: tests/arp/test_neighbor_mac.py [hunk 1]
```diff
@@ -35,15 +36,40 @@ def interfaceConfig(self, duthosts, rand_one_dut_hostname):
...
+
+        portchannel = status["vlan"] if "PortChannel" in status["vlan"] else None
+
+        @contextlib.contextmanager
+        def removeFromPortChannel(duthost, portchannel, intf):
+            try:
+                if portchannel:
+                    duthost.command("sudo config portchannel member del {} {}".format(portchannel, intf))
+                    time.sleep(2)
+                    intfStatus = duthost.show_interface(command="status")["ansible_facts"]["int_status"]
```
Comment (lolyu) [inline tests/arp/test_neighbor_mac.py:56]: better to add a `wait_until` logic here to wait and check.

Comment (byu343) [reply tests/arp/test_neighbor_mac.py]: Done

**Review by lolyu — APPROVED** [review]
LGTM

**Comment by StormLiangMS** [comment]
@byu343 conflicts need to resolve.

**Comment by wsycqyz** [comment]
Please re-run the PR test.

**Review by ZhaohuiS — APPROVED** [review]
LGTM

**Comment by wsycqyz** [comment]
Have triggered to rerun the failed jobs.

**Review by siqbal1986 — APPROVED** [review]

**Comment by StormLiangMS** [comment]
@byu343 code conflict.

**Comment by byu343** [comment]
> @byu343 code conflict.

Code conflict has been fixed

**Pull request merged**